controllers/performanceprofile_controller.go

```diff
@@ -522,3 +534,31 @@ func hasMatchingLabels(performanceprofile *performancev2.PerformanceProfile, mcp
 	}
 	return true
 }
+
+func (r *PerformanceProfileReconciler) getMachineConfigPoolByProfile(profile *performancev2.PerformanceProfile) (*mcov1.MachineConfigPool, error) {
+	nodeSelector := labels.Set(profile.Spec.NodeSelector)
+
+	mcpList := &mcov1.MachineConfigPoolList{}
```

Comment:
It seems you are matching MCPs that use the same node selector. But that might not be accurate. What about going over all matching nodes and using the machineconfiguration.openshift.io/currentConfig annotation on those nodes to collect the MCPs that are in charge of the matched nodes?

Also, in case multiple MCPs match, we must report a degraded condition as well!

Reply:
See also https://github.com/openshift-kni/performance-addon-operators/blob/77d9d8640391f3b54c86cb31c7b1c71a21558778/pkg/profilecreator/mcp.go#L120

Reply:
In general, it should not happen(two MCPs with the same node selector), and should be prevented by MCO validation, but IIRC the MCO does not have such validation, so you are right and I should handle it

Reply:
+1 about Martin's request

Reply:
done, see - https://github.com/openshift-kni/performance-addon-operators/pull/651/files#diff-892a3fda0a96b281dc4e189cc080eeedfa1d9416d5967458ac7651adf7624af3R525